packages/grid/_modules_/grid/hooks/features/filter/useGridFilter.ts

```diff
@@ -174,11 +174,18 @@ export const useGridFilter = (apiRef: GridApiRef, rowsProp: GridRowsProp): void
         };
         return newState;
       });
+
+      // When clicking on "Add Filter" (empty filter) we need not reapply the existing filter
+      if (Object.keys(item).length === 0) {
+        forceUpdate();
```

Comment:
Not sure about this solution, it seems to defer the problem. My assumption was that we shouldn't run the filtering logic when the non-empty filters are the same.

Reply:
We could add an argument to control whether `applyFilters` should be called or not. When deleting the last filter, it's called twice.

Reply:
@m4theushw Yeah, what I was expecting is `applyFilters` to be called as a response to changes in the filterModel, columns or rows **only**.